src/lib/browse/PointAmenityLayerControl.svelte

```diff
@@ -0,0 +1,61 @@
+<script lang="ts">
+  import type { FilterSpecification, MapGeoJSONFeature } from "maplibre-gl";
+  import {
+    isPoint,
+    overwriteCircleLayer,
+    overwriteGeojsonSource,
+  } from "../../maplibre_helpers";
+  import { map } from "../../stores";
+  import { ColorLegend, HelpButton, InteractiveLayer } from "../common";
+  import { Checkbox } from "../govuk";
+  import { colors } from "./colors";
+
+  // This name is used for multiple things:
+  // - The name of a .pmtiles file
+  // - The layer name within that file
+  // - The layer name, for layerZorder
+  // - A color name in colors.ts
+  export let name: string;
+  // Uncapitalized
+  export let singularNoun: string;
+  // Capitalized
+  export let pluralNoun: string;
+
+  export let circleRadius: number;
+
+  // The caller must also fill in the default slot with the contents of a help modal
+
+  // @ts-ignore TODO Also constrain name to exist in the colors type
+  let color = colors[name];
+
+  const layerLoadedCallback = () => {
+    overwriteCircleLayer($map, {
+      id: name,
+      source: name,
+      filter: ["all", isPoint] as FilterSpecification,
+      color: color,
+      radius: circleRadius / 2,
+      // TODO Outline?
+    });
+    show = false;
+  };
+
+  overwriteGeojsonSource($map, name, `https://atip.uk/layers/v1/${name}.geojson`, layerLoadedCallback);
```

Comment:
I am doing it with a callback because I think it needed to be async to handle getting the data,  so this behaves a bit differently to the polygon,  pmtile based layers

Reply:
We don't need a callback; the MapLibre API handles all of this. We just have to add a source before adding the layer that references it. Under the hood, it fetches the data and feeds it to the layer rendering. See https://maplibre.org/maplibre-gl-js/docs/examples/geojson-polygon/ for a simple example

Reply:
Not sure why what I tried before didn't work but it's alright now